district_data_import.py

```diff
@@ -0,0 +1,415 @@
+            'STU_LNAME': district_record['Student Last Name'],
+            'STU_NUMBER': district_record['SystemID'],
+            'SCHOOL': utils.district_school_to_dp_school(district_record['School']),
+            'GRADE': utils.dp_grade_for_district_record(district_record),
+            'OTHER_DATE': TODAY_STR
+        }
+        dp_donor_id_to_other_ids[donor_id].append(other_id)
+        dp_stu_number_to_other_ids[stu_number].append(other_id)
+
+    #End if no donor ID records found
+#End loop over student IDs in district data
+
+# Identify updates to existing donors
+dp_messages_existingdonorrecords = []
+donor_ids_processed = set()
+for stu_number, district_record in district_records.iteritems():
+    district_address = '%s %s %s %s' % (district_record['street'], district_record['city'], district_record['state'], district_record['zip'])
+    district_emails = set([district_record['Parent1Email'].lower(), district_record['Parent2Email'].lower()])
+    district_emails.discard('')
+
+    dp_addresses_by_donor_id = {}
+    dp_emails = set()
+    donor_ids_for_student = set()
+    for other_id in dp_stu_number_to_other_ids[stu_number]:
+        donor_id = dp_studentrecords[other_id]['DONOR_ID']
+        dp_donorrecord = dp_donorrecords[donor_id]
+        donor_ids_for_student.add(donor_id)
+        dp_addresses_by_donor_id[donor_id] = '%s %s %s %s' % (dp_donorrecord['ADDRESS'], dp_donorrecord['CITY'], dp_donorrecord['STATE'], dp_donorrecord['ZIP'])
+        dp_emails.add(dp_donorrecord['EMAIL'].lower())
+        dp_emails.add(dp_donorrecord['SPOUSE_EMAIL'].lower())
+    dp_emails.discard('')
+
+    if donor_ids_processed.issuperset(donor_ids_for_student):
+        continue
+
+    donor_ids_processed.update(donor_ids_for_student)
+
+    if len(donor_ids_for_student) == 1:
+        dp_donorrecord = dp_donorrecords[next(iter(donor_ids_for_student))]
+
+        dp_donorrecord.update({
+            'ADDRESS': district_record['street'],
+            'CITY': district_record['city'],
+            'STATE': district_record['state'],
+            'ZIP': district_record['zip']
+        })
+
+        if district_record['Parent1Email'] and not dp_donorrecord['EMAIL']:
+            dp_donorrecord['EMAIL'] = district_record['Parent1Email']
+
+        if district_record['Parent2Email']:
+            email_field = 'SPOUSE_EMAIL' if district_record['Parent1 Last Name'] else 'EMAIL'
+            if not dp_donorrecord[email_field]:
+                dp_donorrecord[email_field] = district_record['Parent2Email']
+
+        if district_record['GuardianEmail'] and dp_donorrecord['GUARD_EMAIL'].lower().strip() != district_record['GuardianEmail'].lower().strip():
+            dp_donorrecord['GUARD_EMAIL'] = district_record['GuardianEmail']
+
+    # TODO The else case is multiple records. Need to see if anything changed
+
+    if (district_address not in dp_addresses_by_donor_id.values()) or not dp_emails.issuperset(district_emails):
```

Comment:
Ditto. We should put a note here to explain to our future selves what requirement we are trying to meet here.